### PR TITLE
Relative: Fix return of relative calculator if sim fails

### PR DESCRIPTION
**Pull request by Doresic**
In case of simulation fail, the inner relative calculator did not return inner parameters to be added to the inner result. This caused some `ValueError: too many values to unpack (expected 2)` as a tuple was expected.

### CODE REVIEW

File: pypesto/hierarchical/inner_calculator_collector.py [hunk 1]
```diff
@@ -476,7 +476,10 @@ def __call__(
                 snllh += inner_result[GRAD]
 
             all_inner_pars.update(inner_result[X_INNER_OPT])
-            if INNER_PARAMETERS in inner_result:
+            if (
+                INNER_PARAMETERS in inner_result
+                and inner_result[INNER_PARAMETERS] is not None
+            ):
                 interpretable_inner_pars.extend(inner_result[INNER_PARAMETERS])
```
Comment (dweindl) [inline pypesto/hierarchical/inner_calculator_collector.py:481]:
```suggestion
            interpretable_inner_pars.extend(inner_result.get(INNER_PARAMETERS) or [])
```

**Review by dilpath — APPROVED** [review]

**Review by dweindl — APPROVED** [review]
thx :+1:

File: pypesto/hierarchical/inner_calculator_collector.py (hunk 1 above)
Comment (Doresic) [reply pypesto/hierarchical/inner_calculator_collector.py:481]: Ohh that's cleaner, ty :D

Comment (dweindl) [reply pypesto/hierarchical/inner_calculator_collector.py]:
```python
In [1]: {}.get(1, None) or []
Out[1]: []

In [2]: {1:2}.get(1, None) or []
Out[2]: 2
```

Comment (dweindl) [reply pypesto/hierarchical/inner_calculator_collector.py]: Okay, if `get` produces a numpy array, this raises a DeprecationWarning. Not ideal then.

Comment (Doresic) [reply pypesto/hierarchical/inner_calculator_collector.py]: Hmm something else is breaking as well in the notebook, will find a solution fixing both.

Comment (dweindl) [reply pypesto/hierarchical/inner_calculator_collector.py]: fine to keep the original code

Comment (Doresic) [reply pypesto/hierarchical/inner_calculator_collector.py]: Made something in between, not a one-liner but clean enough :D

**Pull request merged**